rclpy/src/rclpy/type_description_service.cpp

```diff
@@ -22,25 +22,12 @@ namespace rclpy
 
 TypeDescriptionService::TypeDescriptionService(Node & node)
 {
-  rcl_ret_t ret = rcl_node_type_description_service_init(node.rcl_ptr());
-  if (RCL_RET_OK != ret) {
-    throw RCLError("Failed to initialize type description service");
-  }
-  rcl_service_t * srv_ptr = nullptr;
-  ret = rcl_node_get_type_description_service(node.rcl_ptr(), &srv_ptr);
+  auto srv_ptr = std::make_shared<rcl_service_t>();
+  rcl_ret_t ret = rcl_node_type_description_service_init(srv_ptr.get(), node.rcl_ptr());
   if (RCL_RET_OK != ret) {
     throw RCLError("Failed to retrieve type description service implementation");
   }
```

Comment:
error message is not correct one.

```suggestion
  if (RCL_RET_OK != ret) {
    throw RCLError("Failed to initialize type description service");
  }
```